### ISSUE TITLE
"File is not included in module tree" does not disappear, code analysis compromised

### ISSUE THREAD

**Issue by blueglyph** [intellij-rust/intellij-rust#9591]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4932-222
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2022.2.3 Ultimate Edition (IU-222.4345.14)
* **Operating system:** Windows 10 10.0
* **Macro expansion:** enabled

## Problem description

Each time a new Rust file is added to a Rust project, IntelliJ shows an invitation to add it to the main file (lib.rs, ...):

![image](https://user-images.githubusercontent.com/31706195/197150441-8a618852-a4df-4036-abdb-25e591f5dfda.png)

After clicking on 'Attach file to lib.rs', the file is correctly attached but the banner remains and the file's semantical analysis is completely off as it ignores all definitions made in the file.

The user can click again on 'Attach file to lib.rs', which adds a new `mod (name)` in lib.rs, producing an error in that file.

The use has to do a File / Invalidate to make the problem disappear.

## Steps to reproduce

- create a library project
- create a new Rust file 'src/example.rs'
=> the IDE creates a blank file and shows the banner
- click on `attach file to lib.rs`
=> `mod example;` is added to the lib.rs file
- re-open 'example.rs' in the editor
=> the banner is still there

- type the following code into example.rs:
```rust
struct Example {}

fn example(a: Example) -> Example {
    a
}
```
- place the cursor on `Example` and Ctrl-B (find declaration)
=> no declaration

![image](https://user-images.githubusercontent.com/31706195/197153509-ccfc1dd2-700b-43a1-b445-342dcb427a56.png)

- compile the code
=> the code correctly compiles (with warnings because of unused code).

**Comment by vlad20012** [comment]
Duplicate #9405, already fixed by #9537, will be released next Monday.

The fix has already been released in the nightly plugin channel. You can try it now (see [instructions](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#install-nightly))

**Comment by blueglyph** [comment]
Thanks, sorry I missed the duplicate.
